### PR TITLE
Add support for having Drupal bootstrapped from other than sites/defa…

### PR DESCRIPTION
**Pull request by badrange**
…ult.

This adds support for multisite projects. build.sh adds a parameter to drush, -l,
that if set will make build.sh new and build.sh update work for legacy projects
that are based on sites.php/have the settings file in a different place than the
normal location - sites/default.

### CODE REVIEW

**Comment by tharna** [comment]
Might be better idea to append the -l parameter to the args on the _drush function so it would apply to all drush commands if needed. And only append it if the multisite_site variable is set.

**Comment by badrange** [comment]
Good point with adding the -l parameter to the _drush function itself. About the only appending if set, I figured that we'd save some lines of code by using `default` if it is not set, that is the default anyway - and it worked fine for a standard project according to my testing.

**Comment by tharna** [comment]
True, less code is more ;) So using default is ok.

**Comment by badrange** [comment]
And about the other thing, if we decide to add -l to the drush command itself, we should also refactor build.sh so it adds `"--root=" + format(self.final_build_dir + self.drupal_subpath),` to all commands too, right?

**Comment by tharna** [comment]
That's a good idea too. For most of the commands I think it's ok, but need to check whether it works with the make command.

**Comment by tharna** [comment]
Doesn't seem to interfere with the make, so yes, root could also be moved to _drush.

**Comment by badrange** [comment]
Hope you have time to take a look at this at some point! :)

File: build.sh [hunk 1]
```diff
@@ -438,12 +436,14 @@ class Maker:
...
+		bootstrap_args = ["--root=" + format(self.final_build_dir + self.drupal_subpath), "-l", self.multisite_site]
 		if quiet:
 			FNULL = open(os.devnull, 'w')
-			return subprocess.call([self.drush] + args, stdout=FNULL, stderr=FNULL) == 0
+			return subprocess.call([self.drush] + bootstrap_args + args, stdout=FNULL, stderr=FNULL) == 0
                 if output:
-                        return subprocess.check_output([self.drush] + args)
-		return subprocess.call([self.drush] + args) == 0
+                        return subprocess.check_output([self.drush] + bootstrap_args + args)
+		print bootstrap_args + args
```
Comment (tharna) [inline build.sh]: Some leftover debug output?

**Comment by badrange** [comment]
Oooff, how embarrassing! 

Question: Did any of us test how these fixes work in a Drupal 8 environment?

**Comment by tharna** [comment]
Good point testing D8 separately @mgalang any chance you got time to test this out on D8?

File: build.sh [hunk 2]
```diff
@@ -497,14 +497,14 @@ class Maker:
                 query = "SELECT name from users WHERE uid=1"
                 uid1_name = self._drush(
```
Comment (tharna) [inline build.sh]: Isn't it actually the opening bracket here that is missing? The first parameter to _drush should be an array of drush commandline parameters.

**Comment by badrange** [comment]
Uh, _blush_ :)

**Pull request merged**